redis-persistence/src/main/java/com/netflix/conductor/redis/dao/RedisExecutionDAO.java

```diff
@@ -784,4 +775,12 @@ private void validate(Task task) {
             throw new ApplicationException(Code.INVALID_INPUT, npe.getMessage(), npe);
         }
     }
+
+    public Set<String> getWorkflowIdSetByCorrelationId(String correlationId) {
```

Comment:
the changes look straightforward, now the question is how good is the functionality that maintains the correlation list. We will need a good set of tests cases to make sure this returns the correct set of workflows to delete
The tests to try 
- single workflow (without subworkflow) 
- nested subworkflows, we support I believe nesting of level 20. If we have a workflow with that much nesting does it delete all the levels 
- Parallel loop with subworkflows 
- Serial loop with subworkflows
- Fork with sub workflow 
- case statement that executes a subworkflow

Reply:
Inside serial loop subworkflows are not supported. checking other use cases as well.

Reply:
other than serial loop, all use cases are validated. corrleationid to workflowid map, captures all the workflows correctly.